apps/web/src/components/EditProject.tsx

```diff
@@ -0,0 +1,160 @@
+import { useEffect, useState } from 'react';
+import { Link, useNavigate, useParams } from 'react-router-dom';
+
+import InputField from './SignUp/inputField';
+
+const EditProject = () => {
+  const { id } = useParams();
+
+  useEffect(() => {
+    const apiBaseUrl = 'http://localhost:3000';
+
+    fetch(`${apiBaseUrl}/api/v1/projects/${id}`)
+      .then((response) => response.json())
+      .then((data) =>
+        setProjectDetails({ ...projectDetails, name: data.name, description: data.description })
+      );
+  }, []);
+
+  const navigate = useNavigate();
+  const [projectDetails, setProjectDetails] = useState({
+    name: '',
+    description: ''
+  });
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [successMessage, setSuccessMessage] = useState('');
+  const [error, setError] = useState('');
+  const [_, setApiError] = useState('');
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setProjectDetails((prev) => ({ ...prev, [e.target.name]: e.target.value }));
+  };
+
+  const clientSideValidation = () => {
+    let isFormValid = true;
+    // checking name field
+    if (!projectDetails.name.trim()) {
+      isFormValid = false;
+      setError('Please input a valid name');
+    }
+    // checking for name length
+    if (projectDetails.name.trim().length < 3) {
+      isFormValid = false;
+      setError('Project name must be at least 3 characters');
+    }
+
+    return isFormValid;
+  };
+
+  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    const base = import.meta.env.VITE_API_URL;
+    setError('');
+    setSuccessMessage('');
+    if (!clientSideValidation()) {
+      return;
+    }
+    setIsSubmitting(true);
+    try {
+      const response = await fetch(`${base}/auth/project/:id`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify(projectDetails)
+      });
+      const data = await response.json();
+      if (response.status === 201) {
```

Comment:
You’re treating this like “create project” (201, localStorage, redirect)

For edit, you probably want:

success message

redirect back to /projects/:id or /projects